File: src/translator/test_translator/turing/rate_5_model.jl
# Translated code start.
using Turing
@model function rate_5_model(n1, n2, k1, k2)
    theta ~ Beta(1, 1)
    k1 ~ Binomial(n1, theta)
    k2 ~ Binomial(n2, theta)
    postpredk1 ~ Binomial(n1, theta)
    postpredk2 ~ Binomial(n2, theta)
    return (postpredk1, postpredk2)
end
# Translated code end.
# Test data.
n1 = 50
n2 = 75
k1 = 30
k2 = 43
model = rate_5_model
arguments = (n1, n2, k1, k2)
addresses = ("theta",)
# Inference.
Turing.Random.seed!(0)
inferred = sample(model(arguments...), IS(), 10_000, progress=false)
println("Inferred:")
weights = exp.(inferred[:lp])
weights = weights / sum(weights)
for address in addresses
    mean = sum(inferred[address] .* weights)
    println(" - $address=$mean")
end

File: src/translator/test_translator/pyro/rate_5_model.py
# Translated code start.
import pyro
import pyro.distributions as dist
def rate_5_model(n1, n2, k1, k2):
    theta = pyro.sample('theta', dist.Beta(1, 1))
    pyro.sample('obs1', dist.Binomial(n1, theta), obs=k1)
    pyro.sample('obs2', dist.Binomial(n2, theta), obs=k2)
    postpredk1 = pyro.sample('postpredk1', dist.Binomial(n1, theta))
    postpredk2 = pyro.sample('postpredk2', dist.Binomial(n2, theta))
    return (postpredk1, postpredk2)
# Translated code end.
import torch
# Test data.
n1 = torch.tensor(10)
n2 = torch.tensor(15)
k1 = torch.tensor(7)
k2 = torch.tensor(8)
model = rate_5_model
arguments = (n1, n2, k1, k2)
addresses = ["theta"]
# Inference.
pyro.set_rng_seed(0)
importance = pyro.infer.Importance(model, num_samples=10_000)
posterior = importance.run(*arguments)
inferred = pyro.infer.EmpiricalMarginal(posterior, sites=addresses)
print("Inferred:")
for i, address in enumerate(addresses):
    print(f" - {address}={inferred.mean[i]}")
